skbio/tree/_tree.py

```diff
@@ -3795,70 +3812,159 @@ def accumulate_to_ancestor(self, ancestor):
-        and extends that method to allow the calculation of total descending
-        branch length of a subset of the tips if requested. The postorder
-        guarantees that the function will always be able to add the descending
-        branch length if the node is not a tip.
+        The metric can be considered as the total amount of evolutionary change across
+        all lineages in the tree.
+
+        This metric is closely related to phylogenetic diversity (PD) in community
+        ecology. When ``include_stem`` is True, it is equivalent to Faith's PD (see
+        :func:`~skbio.diversity.alpha.faith_pd`). However, this method is optimized
+        to handle a single set of nodes, whereas the referred function is optimized
+        to simultaneously calculate for multiple taxon sets (i.e., communities).
 
-        Nodes with no length will have their length set to 0. The root length
-        (if it exists) is ignored.
+        Missing branch lengths will be replaced with 0.
 
         Examples
         --------
         >>> from skbio import TreeNode
-        >>> tr = TreeNode.read(["(((A:.1,B:1.2)C:.6,(D:.9,E:.6)F:.9)G:2.4,"
-        ...                     "(H:.4,I:.5)J:1.3)K;"])
-        >>> tdbl = tr.descending_branch_length()
-        >>> sdbl = tr.descending_branch_length(['A', 'E'])
-        >>> print(round(tdbl, 1), round(sdbl, 1))
-        8.9 2.2
+        >>> tree = TreeNode.read([
+        ...     "(((A:.1,B:1.2)C:.6,(D:.9,E:.6)F:.9)G:2.4,(H:.4,I:.5)J:1.3)K;"])
+        >>> print(tree.ascii_art())
+                                      /-A
+                            /C-------|
+                           |          \-B
+                  /G-------|
+                 |         |          /-D
+                 |          \F-------|
+        -K-------|                    \-E
+                 |
+                 |          /-H
+                  \J-------|
+                            \-I
+
+        Calculate the total branch length of the tree.
+
+        >>> L = tree.total_length()
+        >>> print(round(L, 1))
+        8.9
+
+        Calculate the total branch length connecting three taxa.
+
+        >>> L = tree.total_length(['A', 'E', 'H'])
+        >>> print(round(L, 1))
+        6.3
 
         """
-        self.assign_ids()
-        if tip_subset is not None:
-            all_tips = self.subset()
-            if not set(tip_subset).issubset(all_tips):
-                raise ValueError("tip_subset contains ids that aren't tip " "names.")
-
-            lca = self.lowest_common_ancestor(tip_subset)
-            ancestors = {}
-            for tip in tip_subset:
-                curr = self.find(tip)
-                while curr is not lca:
-                    ancestors[curr.id] = curr.length if curr.length is not None else 0.0
-                    curr = curr.parent
-            return sum(ancestors.values())
+        if kwargs and "tip_subset" in kwargs:
+            nodes = kwargs["tip_subset"]
```

Comment:
Similar concern here regarding potential passing of `tip_subset` and `nodes` at the same time.